### PR TITLE
Remove nose, subclass Django's test runner #9260 

### LINKED ISSUE #9260: django-nose in python 3.10
There are issues with django nose in python 3.10 - specifically an older version of collections is used.  From what I can tell, nose is a dead project, and the maintainers have moved on to nose2.  Further, it seems like django-nose is in maintenance mode along with nose.  I am unsure on whether or not they will upgrade nose or django nose to fix this issue, and regardless, it may be time to select a new test runner or figure out a new way forward with our backend tests.  

One place (there are probably more) that will for sure need to be fixed is the tests __init__.py - where setUpTestPackage is called.  That is not called by the default django test runner - only by nose.  So that will need to be moved/replaced.  The test suite did seem to run with the default django test runner (besides the issues that I noted above that cause database errors because setUpTestPackage is not called) - but there are likely other kinks to work out.

### PR DESCRIPTION
**Pull request by jacobtylerwalls**
### Types of changes

-   [ ] Bugfix (non-breaking change which fixes an issue)
-   [ ] New feature (non-breaking change which adds functionality)
-   [x] Breaking change (fix or feature that would cause existing functionality to not work as expected)

### Description of Change
`nose` is no longer maintained and won't run on Python 3.10. This PR subclasses Django's built-in test runner to apply the custom functions that were being supplied to nose.

### Issues Solved
Closes #9260

### Checklist

-   [x] Unit tests pass locally with my changes
-   [ ] I have added tests that prove my fix is effective or that my feature works
-   [ ] I have added necessary documentation (if appropriate)

#### Ticket Background
*   Sponsored by: Farallon Geographics
*   Found by: @aarongundel
*   Tested by: @jacobtylerwalls
*   Designed by: @

### CODE REVIEW

**Comment by jacobtylerwalls** [comment]
I can follow up with a PR to arches-docs next.

File: tests/test_settings.py [hunk 1]
```diff
@@ -100,11 +99,6 @@
 ENABLE_TWO_FACTOR_AUTHENTICATION = False
 FORCE_TWO_FACTOR_AUTHENTICATION = False
 
-# Tell nose to measure coverage on the 'foo' and 'bar' apps
-NOSE_ARGS = ["--with-coverage", "--nologcapture", "--cover-package=arches", "--verbosity=1", "--cover-erase", "--cover-xml", "-s"]
-
-INSTALLED_APPS = INSTALLED_APPS + ("django_nose",)
-
 DATATYPE_LOCATIONS.append("tests.fixtures.datatypes")
 ELASTICSEARCH_HOSTS = [{"scheme": "http", "host": "localhost", "port": ELASTICSEARCH_HTTP_PORT}]
```
Comment (jacobtylerwalls) [inline tests/test_settings.py:103]: I had to manually add "scheme": "http" over to settings_docker.py to have the tests run on Docker.

**Review by aarongundel — APPROVED** [review]
LGTM

**Comment by jacobtylerwalls** [comment]
Thanks for verifying it works in Docker!

**Pull request merged**